stetho/src/main/java/com/facebook/stetho/dumpapp/plugins/HprofDumperPlugin.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.dumpapp.plugins;
+
+import android.content.Context;
+import android.os.Debug;
+
+import com.facebook.stetho.common.Util;
+import com.facebook.stetho.dumpapp.DumpException;
+import com.facebook.stetho.dumpapp.DumpUsageException;
+import com.facebook.stetho.dumpapp.DumperContext;
+import com.facebook.stetho.dumpapp.DumperPlugin;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.Iterator;
+
+/**
+ * Generates an hprof on the sdcard and returns to the client the absolute path to the file.
+ * <p/>
+ * For convenient usage, see: {@code scripts/hprof_dump.sh}
+ */
+public class HprofDumperPlugin implements DumperPlugin {
+
+  private static final String NAME = "hprof";
+  private final Context mContext;
+
+  public HprofDumperPlugin(Context context) {
+    mContext = context;
+  }
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public void dump(DumperContext dumpContext) throws DumpException {
+    final PrintStream output = dumpContext.getStdout();
+
+    Iterator<String> argsIter = dumpContext.getArgsAsList().iterator();
+    String outputPath = argsIter.hasNext() ? argsIter.next() : null;
+    if (outputPath == null) {
+      usage(output);
+    } else {
+      if ("-".equals(outputPath)) {
+        handlePipeOutput(output);
+      } else {
+        File outputFile = new File(outputPath);
+        if (!outputFile.isAbsolute()) {
+          outputFile = mContext.getFileStreamPath(outputPath);
+        }
+        writeHprof(outputFile);
+        output.println("Wrote to " + outputFile);
```

Comment:
Ack'd, though this is intending to be human readable since a proper scripted approach would hardcode the path and therefore it would know the path that was written if the command simply exited successfully.  Also note that scripts would be much better served by the `"-"` output path, as is the case with `hprof_dump.sh`.  Prefer to keep the language here as-is for human beings...